tsdb/engine/tsm1/bool.go

```diff
@@ -97,19 +100,32 @@ type BooleanDecoder struct {
 // SetBytes initializes the decoder with a new set of bytes to read from.
 // This must be called before calling any other methods.
 func (e *BooleanDecoder) SetBytes(b []byte) {
+	if len(b) == 0 {
+		e.err = fmt.Errorf("BooleanDecoder: initialized with empty data")
+		return
+	}
+
 	// First byte stores the encoding type, only have 1 bit-packet format
 	// currently ignore for now.
 	b = b[1:]
 	count, n := binary.Uvarint(b)
+	if n <= 0 {
+		e.err = fmt.Errorf("BooleanDecoder: invalid count")
+		return
+	}
 
 	e.b = b[n:]
 	e.i = -1
 	e.n = int(count)
 }
 
 func (e *BooleanDecoder) Next() bool {
+	if e.err != nil {
+		return false
+	}
+
 	e.i++
-	return e.i < e.n
+	return e.i/8 < len(e.b) && e.i < e.n
```

Comment:
The `e.i/8 < len(e.b)` shouldn't be necessary.  It also adds a division in a hot section of decoder.